### PR TITLE
Feature/update category selection

### PR DESCRIPTION
**Pull request by cbmackintosh**
## What does this PR do?

- Changes category selection from a long dropdown menu to a more user friendly tag system which allows users to filter the apis by selecting/deselecting one or more tags at the top of the page.
- Refactors change handler methods in the FilterForm component and breaks up the filter method in the App component for improved readability and closer adherence to SRP.

## How should it be tested in the UI?

- Conduct searches by changing form elements in different orders to verify that search results are displaying as expected. Try searching for all the Anime and Government apis with an Auth value of OAuth for example.

### CODE REVIEW

File: src/components/FilterForm/FilterForm.tsx [hunk 1]
```diff
@@ -27,62 +27,79 @@ export class FilterForm extends React.Component<FilterProps> {
...
-      HTTPS: '',
-      Cors: '',
+      Categories: [],
+      Auth: 'all',
+      HTTPS: 'all',
+      Cors: 'all',
     }
   }
 
   handleChange = (e: any):void => {
```
Comment (Jacksonmcguire) [inline src/components/FilterForm/FilterForm.tsx:37]: Sweet refactor on handlechange, looking way cleaner

File: src/components/App/App.tsx [hunk 2]
```diff
@@ -40,42 +40,55 @@ class App extends React.Component<Props> {
...
+    } else {
+      return matchedCards
+    }
+  }
 
-    if (matchingCards.length) {
-      this.setState({ ...this.state, currentApis: matchingCards });
-      return matchingCards;
-    } else return this.state.apiList;
+  filter = (stateObj: FilterState): Api[] => {
```
Comment (Jacksonmcguire) [inline src/components/App/App.tsx:83]: Much cleaner than before, good call breaking those out to their own methods

**Review by Jacksonmcguire — APPROVED** [review]
It's all looking good, couldn't find much to comment on, think it's good to merge.

File: src/components/ApiCard/ApiCard.tsx [hunk 3]
```diff
@@ -12,11 +12,11 @@ export const ApiCard = ({API, Auth, Cors, HTTPS, Category, Description, Link}: A
         <p className="category">{Category}</p>
         <h2>{API}</h2>
         <p>{Description}</p>
-        <p>{Auth + ' Auth'}</p>
-        <p>{Cors + ' Cors'}</p>
-        <p>{HTTPS}</p>
+        <p><strong>Auth:</strong> {Auth.length ? Auth : 'No'}</p>
```
Comment (itsnameissteven) [inline src/components/ApiCard/ApiCard.tsx:15]: I like this logic

File: src/components/App/App.tsx (hunk 2 above)
Comment (itsnameissteven) [reply src/components/App/App.tsx:83]: agreed

**Review by itsnameissteven — APPROVED** [review]
everything looks good. Ill merge right now

**Pull request merged**